### PR TITLE
Implement life system

### PR DESCRIPTION
**Pull request by oblekta**

### CODE REVIEW

File: game/CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,7 @@ set(SYSTEM_SRC
         systems/ItemSystem.h
         systems/items/ReverseGravitySystem.cpp
         systems/items/ReverseGravitySystem.h
-        )
+        systems/LifeSystem.cpp systems/LifeSystem.h)
```
Comment (notzain) [inline game/CMakeLists.txt]: newlines, ')' on a seperate line

File: game/CMakeLists.txt [hunk 2]
```diff
@@ -85,7 +85,7 @@ set(BUILDER_SRC
         builders/IBuilder.h
         builders/CharacterBuilder.cpp
         builders/CharacterBuilder.h
-        )
+        builders/CharacterSpawnGenerator.cpp builders/CharacterSpawnGenerator.h)
```
Comment (notzain) [inline game/CMakeLists.txt]: newlines, ')' on a seperate line

File: game/builders/CharacterSpawnGenerator.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include "CharacterSpawnGenerator.h"
+#include <game/components/CharacterSpawnComponent.h>
+#include <game/components/PositionComponent.h>
+#include <game/exceptions/MissingCharacterSpawnException.h>
+
+std::vector<common::Vector2D<double>> game::builders::CharacterSpawnGenerator::collect(engine::ecs::World& world)
```
Comment (notzain) [inline game/builders/CharacterSpawnGenerator.cpp]:
```cpp
namespace game::builders {
std::vector<common::Vector2D<double>>CharacterSpawnGenerator::collect(engine::ecs::World& world)
}
```

Looks cleaner, easier to edit later, and consistent with the rest of the files.

File: game/systems/LifeSystem.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+#include <engine/common/RNG.h>
+#include <game/builders/CharacterSpawnGenerator.h>
+#include <game/components/BodyComponent.h>
+#include <game/components/DimensionComponent.h>
+#include <game/components/EquipmentComponent.h>
+#include <game/components/InventoryComponent.h>
+#include <game/components/ItemComponent.h>
+#include <game/components/LifeComponent.h>
+#include <game/components/PositionComponent.h>
+#include <iostream>
```
Comment (FabioGNR) [inline game/systems/LifeSystem.cpp]: iostream header not used?

File: game/builders/CharacterSpawnGenerator.cpp (hunk 3 above)
Comment (oblekta) [reply game/builders/CharacterSpawnGenerator.cpp]: Thanks for the reminder!

File: game/CMakeLists.txt (hunk 2 above)
Comment (oblekta) [reply game/CMakeLists.txt]: Of course! *ashamed*

**Comment by FabioGNR** [comment]
Merge this shizzle

**Pull request merged**